### DISCUSSION TITLE
Positron - Quarto - hotkey trouble

### DISCUSSION THREAD

**Discussion by jjcrump3** [posit-dev/positron#5399] [Q&A]
# Positron Version ---------------------------------------------
Positron Version: 2024.11.0 (Universal) build 140
Code - OSS Version: 1.93.0
Commit: e0d844b031f95acbf89f234a2cce2af9b6721f6c
Date: 2024-10-31T14:24:19.540Z
Electron: 30.4.0
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0

# Macbook --------------------
R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.6

# Problem Encountered -------------------
While working in a .qmd file, {r} cell, using the hotkey cmd + shft + M, I expect a pipe operator to be inserted into the cell and to continue my analysis.  However, the result I keep getting is not an inserted |>, but rather this...
<img width="1133" alt="Screenshot 2024-11-17 at 7 54 35 PM" src="https://github.com/user-attachments/assets/71937733-c943-4b78-942d-d3f2c750aaca">

with a "no diagnostics" message below the cell?

I can manually insert the |>, but using hotkeys is a is a better process for me (I am using the visual edit mode, but the problem remains with the source edit option too).  I do not think I accidentally changed a setting, though it is possible.  I am not sure how to fix this issue other than moving back to RStudio, but I prefer using Positron, now that I know it exists.

Thank you for any insight / support with this issue, it is quite annoying.

**Comment by jjcrump3** [comment]
Ah, I need to enable RStudio Keymap in settings it seems.  My apologies, had to read some of the comments very carefully to identify the setting.

https://positron.posit.co/keyboard-shortcuts.html#rstudio-keymap

<img width="1150" alt="image" src="https://github.com/user-attachments/assets/800b6b18-ae2a-4d12-b7bf-32683441cc70">

**Reply by jjcrump3** [reply]
Nevermind, still encountering the same error with cmd + shft + M in a .qmd file

<img width="1150" alt="image" src="https://github.com/user-attachments/assets/c4d472fe-0734-4d7a-96ea-4f0a8d959087">

**Reply by jjcrump3** [reply]
Seems like this experience is related to this issue...
[1805](https://github.com/posit-dev/positron/issues/1805)

**Reply by juliasilge** [reply]
Yep, that's correct! We have some problems currently around visual mode that we are tracking mostly in #1805; let us know if you find different/new types of problems there.

**Comment by kv9898** [comment]
In the meantime it should work in the source mode!

**Reply by jjcrump3** [reply]
Indeed! an opportunity to remember my markdown skillset
